src/data/polynomial/module.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+
+import ring_theory.finiteness
+
+/-!
+# Polynomial module
+
+In this file, we define the polynomial module for an `R`-module `M`, i.e. the `R[X]`-module `M[X]`.
```

Comment:
Please explain design decisions (e.g., why do you need a type synonym) in the module docstring.